test/controllers/rentals_controller_test.rb

```diff
@@ -0,0 +1,198 @@
+require "test_helper"
+
+describe RentalsController do
+
+  def check_response(expected_type:, expected_status: :success)
+    must_respond_with expected_status
+    expect(response.header['Content-Type']).must_include 'json'
+
+    body = JSON.parse(response.body)
+    expect(body).must_be_kind_of expected_type
+    return body
+  end
+
+  describe "create" do
+    before do
+      Customer.create(
+        name: "Evelynn",
+        registered_at: "Wed, 29 Apr 2015 14:54:14 UTC +00:00",
+        address: "13133 111th Dr Se",
+        city: "Monroe",
+        state: "WA",
+        postal_code: "98989",
+        phone: "425 425 44 44",
+        videos_checked_out_count: 0
+      )
+
+      Video.create(
+        title: "Cinderella",
+        overview: "After her father unexpectedly dies, young Ella (Lily James) finds herself at the mercy of her cruel stepmother (Cate Blanchett) and stepsisters, who reduce her to scullery maid. Despite her circumstances, she refuses to despair.",
+        release_date: "2015-03-06",
+        total_inventory: 5,
+        available_inventory: 5
+      )
+
+      @customer = Customer.find_by(name: "Evelynn")
+      @video = Video.find_by(title: "Cinderella")
+
+      @rental_data = {
+        video_id: @video.id,
+        customer_id: @customer.id
+      }
+    end
+
+    it "can create a rental" do
+      expect{post rentals_path, params: @rental_data}.must_differ "Rental.count", 1
+      check_response(expected_type: Hash)
+    end
+
+    it "will return JSON with the following keys: customer_id, video_id, due_date, videos_checked_out_count, available_inventory" do
+      json_fields = ["customer_id", "video_id", "due_date", "videos_checked_out_count", "available_inventory"].sort
+      
+      post rentals_path, params: @rental_data
+      body = check_response(expected_type: Hash)
+
+      rental = Rental.last
+      customer = Customer.find_by(name: "Evelynn")
+      video = Video.find_by(title: "Cinderella")
+
+    
+      expect(body.keys.sort).must_equal json_fields
+      expect(body["customer_id"]).must_equal customer.id
+      expect(body["video_id"]).must_equal video.id
+      expect(body["videos_checked_out_count"]).must_equal customer.videos_checked_out_count
+      expect(body["available_inventory"]).must_equal video.available_inventory
+    end
+
+    it "increase the customer's videos_checked_out_count by one" do
+      expect(@customer.videos_checked_out_count).must_equal 0
+      
+      post rentals_path, params: @rental_data
+      customer = Customer.find_by(name: "Evelynn")
+      expect(customer.videos_checked_out_count).must_equal 1
+    end
```

Comment:
You could make this cleaner by using `must_differ`.

Also, it's a good idea to assert on the response code here as well.

```suggestion
    it "increase the customer's videos_checked_out_count by one" do
      expect do
        post rentals_path, params: @rental_data
      end.must_differ "@customer.refresh.videos_checked_out_count", 1
      
      must_respond_with :ok
    end
```